### ISSUE TITLE
Java SDK does not have option to register workflow name statically

### ISSUE THREAD

**Issue by ftkurt** [temporalio/sdk-java#2122]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when trying to register a Java workflow with a static name. The current option using annotations does not work for our use case. We define proto enums and share them across different apps so they can trigger workflows across apps. For instance, we have:
```java
public interface MyWorkflow {
    @WorkflowMethod(name = "MyStaticWorkflowName")
    void execute();
}
```
We need a way to set a default workflow implementation type name during registration.

**Describe the solution you'd like**
A method to register a workflow with a static name during registration using something like:
```java
WorkflowImplementationOptions options = WorkflowImplementationOptions.newBuilder()
        .setDefaultWorkflowImplementationTypeName(Workflows.WORKFLOWS_SEND_SUBMISSION.name())
        .build();

worker.registerWorkflowImplementationTypes(options, SendSubmissionWorkflowImpl.class);
```
This would allow us to define workflows that can be triggered across different apps with a shared static name.

**Describe alternatives you've considered**
Currently, the alternative is to use the short name of the workflow interface by default. However, this does not meet our needs as we cannot share interfaces across apps written in different languages (Golang, Python). Using annotations alone is insufficient for our scenario.

**Additional context**
Our apps in Golang and Python cannot share Java interfaces, hence the need for a static name registration method.

**Comment by Quinn-With-Two-Ns** [comment]
Would you also need the same functionality for activity names?

**Comment by ftkurt** [comment]
@Quinn-With-Two-Ns yes, that will be useful as well. We are running some activities cross platform as well.
